src/Core/Form/ChoiceProvider/MailMethodChoiceProvider.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Form\ChoiceProvider;
+
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Email\MailOption;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class MailMethodChoiceProvider provides choices for mail methods
+ */
+final class MailMethodChoiceProvider implements FormChoiceProviderInterface
+{
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param ConfigurationInterface $configuration
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(
+        ConfigurationInterface $configuration,
+        TranslatorInterface $translator
+    ) {
+        $this->configuration = $configuration;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getChoices()
+    {
+        $choices = [];
+
+        if (null === $this->configuration->get('_PS_HOST_MODE_')) {
+            $choices[$this->transLabel('Use PHP\'s mail() function (recommended; works in most cases)')] =
```

Comment:
Does it need domains?

Reply:
yup, domains are under `transLabel()`, but translations are working.

Reply:
Watchout with `this->transLabel` not sure the translationbundle tool can detect this :/

Reply:
yeah, i was thinking the same, i guess ill have to rollback to using `trans()`. :/

Reply:
Sad, maybe one day we will have a better way to generate translations :(